packages/material-ui/src/Unstable_TrapFocus/Unstable_TrapFocus.js

```diff
@@ -16,26 +30,46 @@ function Unstable_TrapFocus(props) {
     disableRestoreFocus = false,
     getDoc,
     isEnabled,
+    focusSelectors = [],
     open,
   } = props;
   const ignoreNextEnforceFocus = React.useRef();
+  const lastEvent = React.useRef(null);
```

Comment:
Can be dropped per https://github.com/mui-org/material-ui/pull/21857#discussion_r465225886

Reply:
Disagree on this one. We actually need to know if a user is doing shift+tab vs tab.

Scenarios:

1) Focus is moved to 'sentinelStart'. If a user hits tab we know we need to move them to the first tabbable element
2) Focus is moved to 'sentinelStart'. If a user hits shift+tab we know we need to move them to the last focusable event.

The previous suggestion for providing the position of start/end is nice, I don't see how you can do without accounting for whether it's a shift+tab or tab as landing on 'sentinelStart' doesn't tell you the full picture

Reply:
I can only think of one case were this is relevant:

```jsx
<trap-focus>
  <input>
  <input>
</trap-focus>
<button />
```

The user is on the button the trap focus is mounted but not enabled yet. The user press <kbd>Shift</kbd>+<kbd>Tab</kbd>, it lands on the end sentinel, we should move him to the second input, not the first one.

Reply:
Well, no, my case work too.

Reply:
Hum, ok, we can't access `event.shiftKey` from the focus event, only keydown.

Reply:
DOM:
`
<button />
<trap-focus>
  <input>
  <input>
</trap-focus>

1) assume focus is on <button /> at start
2) user hits tab
3) focus now to first <input />
4) user now hits shift+tab
5) focus now goes backwards onto 'sentientalStart'

at this point, without something telling us a user did shift+tab where would we send them? if all we have to go on is position=start in this case would be sending the user right back to the first <input /> rather than the last <input />